webserver/tyrus/src/main/java/io/helidon/webserver/tyrus/TyrusSupport.java

```diff
@@ -0,0 +1,300 @@
+        private Set<ServerEndpointConfig> endpointConfigs = new HashSet<>();
+
+        private Builder() {
+        }
+
+        /**
+         * Register an endpoint class.
+         *
+         * @param endpointClass The class.
+         * @return The builder.
+         */
+        public Builder register(Class<?> endpointClass) {
+            endpointClasses.add(endpointClass);
+            return this;
+        }
+
+        /**
+         * Register an endpoint config.
+         *
+         * @param endpointConfig The endpoint config.
+         * @return The builder.
+         */
+        public Builder register(ServerEndpointConfig endpointConfig) {
+            endpointConfigs.add(endpointConfig);
+            return this;
+        }
+
+        @Override
+        public TyrusSupport build() {
+            // Create container and WebSocket engine
+            TyrusServerContainer serverContainer = new TyrusServerContainer(endpointClasses) {
+                private final WebSocketEngine engine =
+                        TyrusWebSocketEngine.builder(this).build();
+
+                @Override
+                public void register(Class<?> endpointClass) {
+                    throw new UnsupportedOperationException("Use TyrusWebSocketEngine for registration");
+                }
+
+                @Override
+                public void register(ServerEndpointConfig serverEndpointConfig) {
+                    throw new UnsupportedOperationException("Use TyrusWebSocketEngine for registration");
+                }
+
+                @Override
+                public WebSocketEngine getWebSocketEngine() {
+                    return engine;
+                }
+            };
+
+            // Register classes with context path "/"
+            WebSocketEngine engine = serverContainer.getWebSocketEngine();
+            endpointClasses.forEach(c -> {
+                try {
+                    // Context path handled by Helidon based on app's routes
+                    engine.register(c, "/");
+                } catch (DeploymentException e) {
+                    throw new RuntimeException(e);
+                }
+            });
+            endpointConfigs.forEach(c -> {
+                try {
+                    // Context path handled by Helidon based on app's routes
+                    engine.register(c, "/");
+                } catch (DeploymentException e) {
+                    throw new RuntimeException(e);
+                }
+            });
+
+            // Create TyrusSupport using WebSocket engine
+            return new TyrusSupport(serverContainer.getWebSocketEngine(), endpointClasses, endpointConfigs);
+        }
+    }
+
+    /**
+     * Creates executor service for Websocket in MP. No executor for SE.
+     *
+     * @return Executor service or {@code null}.
+     */
+    private static ExecutorService createExecutorService() {
+        if (HelidonFeatures.flavor() == HelidonFlavor.MP && DEFAULT_THREAD_POOL.get() == null) {
```

Comment:
I do not think the `.flavor()` method is good. We do not control the time when the flavor is configured and this is not the intention (`HelidonFeatures` task is to list features of Helidon, not to handle the flavor of Helidon).
Also the approach for thread pools is to use one of the existing `ThreadPoolSupplier` classes, that already handle lazy initiialization.
If you do not need an executor in SE, than why is this method in this class? It should be in the CDI extension for WebSockets, and the `TyrusSupport` class should only provider features to handle the SE model (with capability to build the MP model on top of it).